app/client/client.js

```diff
@@ -0,0 +1,213 @@
+const { RoleMessage } = require('../models')
+
+const applicationConfig = require('../../config/application')
+
+const ACTIVITY_CAROUSEL_INTERVAL = 60 * 1000
+const COMMAND_DELETE_MESSAGES_TIMEOUT = 10 * 1000
+
+require('../extensions') // Extend Discord.js structures before the client's collections get instantiated.
+
+class NSadminClient extends CommandoClient {
+  constructor (options = {}) {
+    if (typeof options.commandPrefix === 'undefined') {
+      options.commandPrefix = applicationConfig.defaultPrefix
+    }
+    if (typeof options.owner === 'undefined') {
+      options.owner = applicationConfig.owner
+    }
+    if (typeof options.invite === 'undefined') {
+      options.invite = applicationConfig.invite
+    }
+    if (!options.partials) {
+      options.partials = []
+    }
+    if (!options.partials.some(partial => partial === 'REACTION')) {
+      options.partials.push('REACTION')
+    }
+    if (!options.partials.some(partial => partial === 'MESSAGE')) {
+      options.partials.push('MESSAGE')
+    }
+    if (!options.partials.some(partial => partial === 'USER')) {
+      options.partials.push('USER')
+    }
+    super(options)
+
+    this.currentActivity = 0
+
+    this.registry
+      .registerGroup('admin', 'Admin')
+      .registerGroup('settings', 'Settings')
+      .registerGroup('main', 'Main')
+      .registerGroup('miscellaneous', 'Miscellaneous')
+      .registerGroup('bot', 'Bot')
+      .registerGroup('tickets', 'Tickets')
+      .registerDefaultGroups()
+      .registerDefaultTypes()
+      .registerDefaultCommands({
+        commandState: true,
+        unknownCommand: false,
+        ping: true,
+        help: true,
+        eval: true,
+        prefix: true
+      })
+      .registerTypesIn(path.join(__dirname, '../types'))
+      .registerCommandsIn(path.join(__dirname, '../commands'))
+
+    this.once('ready', this.ready.bind(this))
+  }
+
+  async ready () {
+    await this.setProvider(new NSadminProvider())
+
+    const mainGuildId = process.env.NODE_ENV === 'production'
+      ? applicationConfig.productionMainGuildId
+      : applicationConfig.developmentMainGuildId
+    this.mainGuild = this.guilds.cache.get(mainGuildId)
+
+    this.ticketsController = new TicketsController(this)
+    await this.ticketsController.init()
+
+    this.bindEvent('channelDelete')
+    this.bindEvent('commandError')
+    this.bindEvent('commandPrefixChange')
+    this.bindEvent('commandRun')
+    this.bindEvent('commandStatusChange')
+    this.bindEvent('emojiDelete')
+    this.bindEvent('groupStatusChange')
+    this.bindEvent('guildCreate')
+    this.bindEvent('guildMemberAdd')
+    this.bindEvent('message')
+    this.bindEvent('messageDelete')
+    this.bindEvent('messageDeleteBulk')
+    this.bindEvent('messageReactionAdd')
+    this.bindEvent('messageReactionRemove')
+    this.bindEvent('roleDelete')
+
+    if (applicationConfig.apiEnabled) {
+      this.ws = new WebSocketManager(this)
```

Comment:
This overrides discord.js Client's ws variable: change to different name.

nsadminWs? 🤮